### PR TITLE
Character api refactor

### PR DESCRIPTION
**Pull request by spaceplesiosaur**
#### What's this PR do?

This refactors the characterBox.js page and character fetch API into small, clean functions, and moves the fetches into a single, dynamic function called from apiCalls.js.

This is to improve testing and readability.

#### Where should the reviewer start?  
You can just look at apiCalls and CharacterBox.js.  The fakeCharacters file was also updated, but it was just to remove an irrelevant key-value pair so you can ignore it.  The snapshots were updated as well.

#### How should this be manually tested? 

You've already checked it over, but you can pull down and make sure the tests still pass and the app still runs if you'd like.  They are working for me.

#### What are the relevant tickets?

This addresses the following tasks on our Trello board on the Refactor card
- Refacter CharacterBox.js
-Refactor Character Fetch

### CODE REVIEW

File: src/components/CharacterBox/CharacterBox.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import React, { Component } from 'react'
 import CharacterCard from '../CharacterCard/CharacterCard'
+import fakeCharacters from '../../mockData/fakeCharacters.js'
```
Comment (RayRedGoose) [inline src/components/CharacterBox/CharacterBox.js]: We don't use this dataset in this component, maybe it is better to delete it?

File: src/components/CharacterBox/CharacterBox.js [hunk 2]
```diff
@@ -10,50 +12,52 @@ export default class CharacterBox extends Component {
...
+  nestedCharacterFetch = (promise) => {
+    const name = promise.name;
+    return Promise.all([this.speciesFetch(promise.species), this.planetFetch(promise.homeworld), this.filmsFetch(promise.films)])
+      .then(info => {
+        return {name, ...info[0], ...info[1], ...info[2]}
+      })
+        .then(characterStats => {
+          this.setState({characters: [...this.state.characters, characterStats]})
         })
+  }
```
Comment (RayRedGoose) [inline src/components/CharacterBox/CharacterBox.js]: I made small changes to this function. I guess it looks cleaner

Screen Shot 2019-12-08 at 8 52 39 PM

**Review by RayRedGoose — APPROVED** [review]
This is incredible refactoring of this big fetch function. Now it looks perfect!

File: src/components/CharacterBox/CharacterBox.js (hunk 2 above)
Comment (spaceplesiosaur) [reply src/components/CharacterBox/CharacterBox.js]: ooh nice destructuring!

File: src/components/CharacterBox/CharacterBox.js (hunk 1 above)
Comment (spaceplesiosaur) [reply src/components/CharacterBox/CharacterBox.js]: Good catch, just removed it

**Pull request merged**